cc/testing/evaluation/algorithms_with_insufficient_noise.h

```diff
@@ -0,0 +1,124 @@
+#include <iostream>
+#include <map>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "absl/memory/memory.h"
+#include "absl/random/distributions.h"
+
+#include "algorithms/algorithm.h"
+#include "algorithms/numerical-mechanisms.h"
+#include "algorithms/numerical-mechanisms-testing.h"
+#include "algorithms/util.h"
+#include "base/statusor.h"
+#include "proto/data.pb.h"
+#include "testing/sequence.h"
+#include "testing/stochastic_tester.h"
+
+namespace differential_privacy {
+
+namespace testing {
+
+// Runs the Stochastic Tester on a continuous range of three algorithms (Count, 
+// BoundedSum, and BoundedMean) in which the privacy protection claimed exceeds 
+// the amount of noise applied. Sends results of each run to an output file.
+
+template <typename T>
+class CountWithInsufficientNoise : public differential_privacy::Count<T> {
+ public:
+  CountWithInsufficientNoise(double epsilon, double ratio)
+    : Count<T>(epsilon,
+    LaplaceMechanism::Builder()
+      .SetEpsilon(epsilon)
+      .SetL0Sensitivity(1)
+      .SetLInfSensitivity(1)
+      .Build()
+      .ValueOrDie()),
+      ratio_(ratio) {} 
+  double GetEpsilon() const override { return Algorithm<T>::GetEpsilon()
+    * ratio_; }
+  private:
+    double ratio_;
+};
+
+template <typename T, typename std::enable_if<std::is_integral<T>::value || 
+	std::is_floating_point<T>::value>::type* = nullptr>
+class SumWithInsufficientNoise : public differential_privacy::BoundedSum<T> {
+ public:
+  SumWithInsufficientNoise(double epsilon, T lower, T upper,
+    std::unique_ptr<LaplaceMechanism::Builder> builder, double ratio)
+    : BoundedSum<T>(epsilon, lower, upper, 1, 1, std::move(builder),
+    nullptr, nullptr), ratio_(ratio) {} // set sensitivity values to 1
+// Overrides epsilon such that amount of noise applied is only a fraction of
+// what privacy protection claimed
+  double GetEpsilon() const override { return Algorithm<T>::GetEpsilon() 
+   * ratio_; } 
+ private: 
+  double ratio_;
+};
+
+template <typename T, typename std::enable_if<std::is_integral<T>::value || 
+std::is_floating_point<T>::value>::type* = nullptr>
+class MeanWithInsufficientNoise : public differential_privacy::BoundedMean<T> {
+ public:
+  MeanWithInsufficientNoise(double epsilon, T lower, T upper,
+    std::unique_ptr<LaplaceMechanism::Builder> builder, double ratio)
+    : BoundedMean<T>(epsilon, lower, upper, 1, 1, std::move(builder),
+    LaplaceMechanism::Builder()
+      .SetEpsilon(epsilon)
+      .SetL0Sensitivity(1)
+      .SetLInfSensitivity(1)
+      .Build()
+      .ValueOrDie(),
+    LaplaceMechanism::Builder()
+      .SetEpsilon(epsilon)
+      .Build()
+      .ValueOrDie(), nullptr), ratio_(ratio) {}
+    double GetEpsilon() const override { return Algorithm<T>::GetEpsilon()
+     * ratio_; }
+ private:
+  double ratio_;
+};
+
+bool RunStochasticTester(double ratio, double num_datasets,
```

Comment:
There should probably be comments explaining what each of these classes represent, functions do, what the arguments mean, and what callers should expect.

Reply:
Included in ReadMe